python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -14,48 +14,76 @@
-                           "openvino": False, "pot": False, "bf16": False},
-        "None_bf16_noipex": {"inc": False, "ipex": False, "onnxruntime": False,
-                             "openvino": False, "pot": False, "bf16": True},
-        "None_bf16_ipex": {"inc": False, "ipex": True, "onnxruntime": False,
-                           "openvino": False, "pot": False, "bf16": True},
-        "None_int8_noipex": {"inc": True, "ipex": False, "onnxruntime": False,
-                             "openvino": False, "pot": False, "bf16": False},
-        "jit_fp32_noipex": {"inc": False, "ipex": False, "onnxruntime": False,
-                            "openvino": False, "pot": False, "bf16": False},
-        "jit_fp32_ipex": {"inc": False, "ipex": True, "onnxruntime": False,
-                          "openvino": False, "pot": False, "bf16": False},
-        "jit_bf16_noipex": {"inc": False, "ipex": False, "onnxruntime": False,
-                            "openvino": False, "pot": False, "bf16": True},
-        "jit_bf16_ipex": {"inc": False, "ipex": True, "onnxruntime": False,
-                          "openvino": False, "pot": False, "bf16": True},
-        "onnxruntime_fp32_noipex": {"inc": False, "ipex": False, "onnxruntime": True,
-                                    "openvino": False, "pot": False, "bf16": False},
-        "onnxruntime_int8_noipex_qlinear": {"inc": True, "ipex": False, "onnxruntime": True,
-                                            "openvino": False, "pot": False, "bf16": False},
-        "onnxruntime_int8_noipex_integer": {"inc": True, "ipex": False, "onnxruntime": True,
-                                            "openvino": False, "pot": False, "bf16": False},
-        "openvino_fp32_noipex": {"inc": False, "ipex": False, "onnxruntime": False,
-                                 "openvino": True, "pot": False, "bf16": False},
-        "openvino_int8_noipex": {"inc": False, "ipex": False, "onnxruntime": False,
-                                 "openvino": True, "pot": True, "bf16": False},
+        "original": AccelerationOption(),
+        "None_fp32_ipex": AccelerationOption(ipex=True),
+        "None_bf16": AccelerationOption(bf16=True),
+        "None_bf16_ipex": AccelerationOption(bf16=True, ipex=True),
+        "None_int8": AccelerationOption(inc=True),
+        "jit_fp32": AccelerationOption(jit=True),
+        "jit_fp32_ipex": AccelerationOption(jit=True, ipex=True),
+        "jit_fp32_ipex_clast": AccelerationOption(jit=True, ipex=True,
+                                                  channels_last=True),
+        "openvino_fp32": AccelerationOption(openvino=True),
+        "openvino_int8": AccelerationOption(openvino=True, inc=True),
+        "onnxruntime_fp32": AccelerationOption(onnxtunrime=True),
+        "onnxruntime_int8_qlinear": AccelerationOption(onnxruntime=True, inc=True),
+        "onnxruntime_int8_integer": AccelerationOption(onnxruntime=True, inc=True),
```

Comment:
what's the difference between the above two?

Reply:
These are 2 common method we use for onnxruntime quantization. It is hard to decide which one is better since it varies from case to case. the method name (i.e. qlinear, integer) will be parsed in later codes, this dictionary only states the dependencies they need ( they have same dependencies requirement)

Reply:
As mentioned in the comments below, this can be confusing and hard to maintain in the long run (by parsing the string name); add another attribute (such as `method`) instead

Reply:
done